src/rust/src/rbackground.rs

```diff
@@ -281,10 +281,28 @@ impl RBackgroundPool {
     }
 
     pub fn lease(&self) -> RResult<RBackgroundHandler> {
-        if let Some(handle) = self.pool.lock()?.pop_front() {
-            Ok(handle)
-        } else {
-            RBackgroundHandler::new()
+        let mut sleep_time_us: u64 = 200;
+        loop {
+            let mut pool = self.pool.lock()?;
+            let cap = self.cap.lock()?;
+            if let Some(handle) = pool.pop_front() {
+                break Ok(handle);
+            } else {
+                if *cap < 1 {
+                    rerr()
+                        .plain("cannot run backround R process with zero capacity")
+                        .hint("try pl$set_global_rpool_cap(4)")?;
+                }
+                if pool.len() < *cap {
+                    break RBackgroundHandler::new();
+                } else {
```

Comment:
May be the pool is guaranteed to be empty when line 296 is reached, as `pop_front` fails to get an element?